datastore/importers/nmdb-import-systeminfo/Parser.cpp

```diff
@@ -0,0 +1,167 @@
+        qi::lit("DHCP Enabled:") >> (qi::lit("Yes") | qi::lit("No"));
+
+    networkCardStatus =
+        +qi::ascii::graph;
+
+    network_card =
+        ('[' >> qi::lexeme[+qi::char_("0-9")] >> ']' >> qi::lit(':') >> networkCardName[(pnx::bind(&Parser::addInterface, this, qi::_1))] >> qi::eol >> "Connection Name: " > networkCardConnectionName[(pnx::bind(&Parser::addIfaceConnectName, this, qi::_1))] >> qi::eol >> -(dhcpEnabledStatus >> qi::eol) >> -("DHCP Server: " >> dhcpServer >> qi::eol) >> -("IP address(es)" >> qi::eol > +('[' >> qi::lexeme[+qi::char_("0-9")] >> "]: " >> ipAddr[(pnx::bind(&Parser::addIfaceIp, this, qi::_1))] >> qi::eol)) >> -("Status: " >> networkCardStatus[(pnx::bind(&Parser::setIfaceStatus, this, qi::_1))] >> qi::eol));
+    ;
+    network_cards =
+        qi::lit("Network Card(s):") >> +qi::ascii::print > qi::eol > +network_card;
+
+    hyper_v =
+        ("Hyper-V Requirements: " > token > qi::eol);
+
+    systeminfo =
+        host_name > os_name > os_version > os_manufacturer > os_configuration > *(qi::char_ - qi::lit("System Manufacturer:")) > system_manufacturer > system_model > system_type > *(qi::char_ - qi::lit("Domain:")) > domain[pnx::bind(&Parser::setDomain, this, qi::_1)] > *(qi::char_ - qi::lit("Hotfix")) > hotfixes > network_cards > hyper_v;
+    ;
+    ignoredLine =
+        (qi::ascii::print > -qi::eol) | +qi::eol;
+
+    // Allows for error handling and debugging of qi.
+    BOOST_SPIRIT_DEBUG_NODES(
+        (start)(systeminfo)(host_name)(os_name)(os_version)(os_manufacturer)(os_configuration)(system_manufacturer)(system_model)(system_type)(domain)(hotfixes)(hotfix)(network_cards)(network_card)(dhcpServer)(dhcpEnabledStatus)(ipAddressLine)(networkCardName)(networkCardConnectionName)(networkCardStatus));
+}
+
+// =============================================================================
+// Parser helper methods
+// =============================================================================
+
+void Parser::setDomain(const std::string &_string)
+{
+    curDomain = _string;
+}
+
+void Parser::addHotfix(const std::string &_string)
+{
+    data.hotfixes.push_back(_string);
+}
+
+void Parser::addInterface(const std::string &_name)
+{
+    nmdo::Interface iface;
+    iface.setName(_name);
+    curIfaceName = iface.getName();
+    data.network_cards[curIfaceName] = iface;
+}
+
+void Parser::addIfaceConnectName(const std::string &_connectionname)
+{
+    auto &iface{data.network_cards[curIfaceName]};
+    if (_connectionname.find("Wi-Fi") != std::string::npos || _connectionname.find("Bluetooth") != std::string::npos)
+    {
+        iface.setMediaType(_connectionname);
+    }
+    iface.setUp();
+    iface.setDescription(_connectionname);
+}
+
+void Parser::addIfaceIp(nmdo::IpAddress &_ipAddr)
+{
+    auto &iface{data.network_cards[curIfaceName]};
+    std::string fqdn = data.devInfo.getDeviceId() + '.' + curDomain;
+    _ipAddr.addAlias(fqdn, "from systeminfo");
+    data.os.setIpAddr(_ipAddr);
```

Comment:
Couple of thoughts:
1. This overrides the IP address to whatever is last processed.  This is less than ideal as the last processed is probably an IPv6 address that isn't really useful (e.g., `fe80:...`).
2. It may not be the primary interface.
3. No interface may have an IP (e.g., if systeminfo is ran while the network is down).

All of this is less than ideal.  Please set up some time to discuss this as it may take reworking the `OperatingSystem` object.